src/Adapter/Configuration.php

```diff
@@ -395,4 +400,20 @@ public function deleteFromContext(string $key, ShopConstraint $shopConstraint):
             !empty($shopId) ? $shopId->getValue() : null
         );
     }
+
+    /**
+     * @return ShopConstraint
+     */
+    private function buildShopConstraintFromContext(): ShopConstraint
+    {
+        @trigger_error(
+            'Not specifying the optional ShopConstraint parameter is deprecated since version 1.7.8.0',
```

Comment:
That's the only thing I am troubled with in this PR @eternoendless 

We discussed that in the next 8.0 version we would probably got rid of this extra parameter and instead the `Configuration` would become stateful (one configuration is fetched for a specific ShopConstraint), so this future idea seems more stable (for once having a stateful shop context, but specific to one service seems like the cleaner approach)

However this deprecation message goes against this idea since it implies that every future calls of `Configuration::get` will have to provide this extra parameter, meaning we also have to provide a default value everywhere...

TLDR: I'm not against this deprecation message but IMHO it should rather be a message stating that this parameter is gonna be removed in the next version, not become mandatory

Reply:
The new Configuration will belong to the Core namespace, this `Adapter\Configuration` will most likely delegate to `Core\ConfigurationRepository` to get the appropriate `Core\Configuration`

Reply:
This depreciation fills the dev log very quickly. Can we get rid of it?
![obraz](https://user-images.githubusercontent.com/5007456/225288371-89bb4d5a-ef70-41c7-8af3-7a0dad1ea281.png)